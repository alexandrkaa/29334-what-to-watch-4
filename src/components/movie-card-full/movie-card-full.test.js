import React from 'react';
// import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import MovieCardFull from './movie-card-full.jsx';

import ShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ShallowRenderer();

const mockStore = configureStore([]);

const mockData = {
  titleMovie: {
    id: 4,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    movieDate: `2000`,
    movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
    movieDirector: `Wes Anderson`,
    movieGenre: `Thriller`,
    movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
    movieRatingCount: `349 ratings`,
    movieRatingLevel: `Bad`,
    movieRatingScore: `3`,
    movieStarring: [`Jude Law`, `Willem Dafoe`],
    title: `Dardjeeling Limited`,
    moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `#000000`,
    movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isFavorite: false,
  },

  moviesList: [
    {
      id: 4,
      image: `img/we-need-to-talk-about-kevin.jpg`,
      movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
      movieDate: `2000`,
      movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
      movieDirector: `Wes Anderson`,
      movieGenre: `Thriller`,
      movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
      movieRatingCount: `349 ratings`,
      movieRatingLevel: `Bad`,
      movieRatingScore: `3`,
      movieStarring: [`Jude Law`, `Willem Dafoe`],
      title: `Dardjeeling Limited`,
      moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      backgroundColor: `#000000`,
      movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      isFavorite: false,
    },
    {
      id: 3,
      image: `img/aviator.jpg`,
      movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
      movieDate: `2008`,
      movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
      movieDirector: `Christopher Nolan`,
      movieGenre: `Kids & Family`,
      movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
      movieRatingCount: `272 ratings`,
      movieRatingLevel: `Normal`,
      movieRatingScore: `2`,
      movieStarring: [`Jude Law`, `Willem Dafoe`],
      title: `Johnny English`,
      moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      backgroundColor: `#000000`,
      movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      isFavorite: false,
    }
  ],
  movieGenres: [
    `Thriller`,
    `Kids & Family`
  ]
};

const movie = {
  id: 4,
  image: `img/we-need-to-talk-about-kevin.jpg`,
  movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
  movieDate: `2000`,
  movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
  movieDirector: `Wes Anderson`,
  movieGenre: `Thriller`,
  movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
  movieRatingCount: `349 ratings`,
  movieRatingLevel: `Bad`,
  movieRatingScore: `3`,
  movieStarring: [`Jude Law`, `Willem Dafoe`],
  title: `Dardjeeling Limited`,
  moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  movieRunTime: 9900,
  backgroundColor: `#000000`,
  movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isFavorite: false,
};

const similarMovies = [
  {
    id: 5,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    movieDate: `2300`,
    movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
    movieDirector: `Wes Anderson`,
    movieGenre: `Thriller`,
    movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
    movieRatingCount: `349 ratings`,
    movieRatingLevel: `Bad`,
    movieRatingScore: `3`,
    movieStarring: [`Jude Law`, `Willem Dafoe`],
    title: `Dardjeeling Limited`,
    moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    movieRunTime: 9900,
    backgroundColor: `#000000`,
    movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isFavorite: false,
  },
  {
    id: 6,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    movieDate: `2200`,
    movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
    movieDirector: `Wes Anderson`,
    movieGenre: `Thriller`,
    movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
    movieRatingCount: `349 ratings`,
    movieRatingLevel: `Bad`,
    movieRatingScore: `3`,
    movieStarring: [`Jude Law`, `Willem Dafoe`],
    title: `Dardjeeling Limited`,
    moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    movieRunTime: 9900,
    backgroundColor: `#000000`,
    movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isFavorite: false,
  },
  {
    id: 7,
    image: `img/we-need-to-talk-about-kevin.jpg`,
    movieBackground: `img/bg-the-grand-budapest-hotel.jpg`,
    movieDate: `2100`,
    movieDescription: `In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.`,
    movieDirector: `Wes Anderson`,
    movieGenre: `Thriller`,
    movieImage: `img/the-grand-budapest-hotel-poster.jpg`,
    movieRatingCount: `349 ratings`,
    movieRatingLevel: `Bad`,
    movieRatingScore: `3`,
    movieStarring: [`Jude Law`, `Willem Dafoe`],
    title: `Dardjeeling Limited`,
    moviePreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    movieRunTime: 9900,
    backgroundColor: `#000000`,
    movieVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isFavorite: false,
  }
];

const mockTabs = [
  {
    id: `TAB1`,
    name: `Tab1`,
    isActive: true,
  },
  {
    id: `TAB2`,
    name: `Tab2`,
    isActive: false,
  },
  {
    id: `TAB3`,
    name: `Tab3`,
    isActive: false,
  },
];

it(`<MovieCardFull /> should movie full card page to match snapshot`, () => {
  const {titleMovie, moviesList, movieGenres} = mockData;
  const store = mockStore({
    MOVIE_DATA: {
      titleMovie,
      moviesList,
      movieGenres,
      loadingMovies: false,
      loadingMoviesError: false,
    },
    COMMENTS_DATA: {
      loadingComments: false,
      loadingCommentsError: false,
      moviesComments: {3: [
        {
          id: 1,
          user: {
            id: 17,
            name: `Emely`
          },
          rating: 2.9,
          comment: `Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.`,
          date: `2020-07-15T12:25:15.535Z`
        }
      ]},
      postCommentInProgress: false,
      postCommentError: false,
      postCommentSuccess: false,
    },
    MOVIE: {
      activeGenre: `All genres`,
      moviesRenderLimit: 8,
    },
    USER: {
      authorizationStatus: `AUTH`,
      userData: {
        id: 1,
        email: `qwe@qwe.ru`,
        name: `qwe`,
        avatarUrl: `/img/avatar.jpg`
      }
    }
  });
  const onMovieTitleClick = jest.fn();
  // const tree = renderer
  //   .create(
  //       <Provider store={store}>
  //         <BrowserRouter>
  //           <MovieCardFull
  //             movie={movie}
  //             similarMovies={similarMovies}
  //             activeItem={`TAB1`}
  //             onMovieTitleClick={onMovieTitleClick}
  //             tabs={mockTabs}
  //             history={{}}
  //             isAuthorized={true}
  //           />
  //         </BrowserRouter>
  //       </Provider>,
  //       {
  //         createNodeMock: () => {
  //           return {};
  //         }
  //       }
  //   )
  //   .toJSON();
  renderer
    .render(
        <Provider store={store}>
          <BrowserRouter>
            <MovieCardFull
              movie={movie}
              similarMovies={similarMovies}
              activeItem={`TAB1`}
              onMovieTitleClick={onMovieTitleClick}
              tabs={mockTabs}
              history={{}}
              isAuthorized={true}
            />
          </BrowserRouter>
        </Provider>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    );
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
