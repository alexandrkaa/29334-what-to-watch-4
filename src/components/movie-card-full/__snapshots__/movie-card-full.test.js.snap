// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCardFull /> should movie full card page to match snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <BrowserRouter>
    <Memo(Connect(WithActiveItemHoc))
      activeItem="TAB1"
      history={Object {}}
      isAuthorized={true}
      movie={
        Object {
          "backgroundColor": "#000000",
          "id": 4,
          "image": "img/we-need-to-talk-about-kevin.jpg",
          "isFavorite": false,
          "movieBackground": "img/bg-the-grand-budapest-hotel.jpg",
          "movieDate": "2000",
          "movieDescription": "In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.",
          "movieDirector": "Wes Anderson",
          "movieGenre": "Thriller",
          "movieImage": "img/the-grand-budapest-hotel-poster.jpg",
          "moviePreview": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
          "movieRatingCount": "349 ratings",
          "movieRatingLevel": "Bad",
          "movieRatingScore": "3",
          "movieRunTime": 9900,
          "movieStarring": Array [
            "Jude Law",
            "Willem Dafoe",
          ],
          "movieVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
          "title": "Dardjeeling Limited",
        }
      }
      onMovieTitleClick={[MockFunction]}
      similarMovies={
        Array [
          Object {
            "backgroundColor": "#000000",
            "id": 5,
            "image": "img/we-need-to-talk-about-kevin.jpg",
            "isFavorite": false,
            "movieBackground": "img/bg-the-grand-budapest-hotel.jpg",
            "movieDate": "2300",
            "movieDescription": "In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.",
            "movieDirector": "Wes Anderson",
            "movieGenre": "Thriller",
            "movieImage": "img/the-grand-budapest-hotel-poster.jpg",
            "moviePreview": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "movieRatingCount": "349 ratings",
            "movieRatingLevel": "Bad",
            "movieRatingScore": "3",
            "movieRunTime": 9900,
            "movieStarring": Array [
              "Jude Law",
              "Willem Dafoe",
            ],
            "movieVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "title": "Dardjeeling Limited",
          },
          Object {
            "backgroundColor": "#000000",
            "id": 6,
            "image": "img/we-need-to-talk-about-kevin.jpg",
            "isFavorite": false,
            "movieBackground": "img/bg-the-grand-budapest-hotel.jpg",
            "movieDate": "2200",
            "movieDescription": "In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.",
            "movieDirector": "Wes Anderson",
            "movieGenre": "Thriller",
            "movieImage": "img/the-grand-budapest-hotel-poster.jpg",
            "moviePreview": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "movieRatingCount": "349 ratings",
            "movieRatingLevel": "Bad",
            "movieRatingScore": "3",
            "movieRunTime": 9900,
            "movieStarring": Array [
              "Jude Law",
              "Willem Dafoe",
            ],
            "movieVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "title": "Dardjeeling Limited",
          },
          Object {
            "backgroundColor": "#000000",
            "id": 7,
            "image": "img/we-need-to-talk-about-kevin.jpg",
            "isFavorite": false,
            "movieBackground": "img/bg-the-grand-budapest-hotel.jpg",
            "movieDate": "2100",
            "movieDescription": "In the 1930s, the Grand Budapest Hotel is a popula…y boy, becomes Gustave&apos;s friend and protege.",
            "movieDirector": "Wes Anderson",
            "movieGenre": "Thriller",
            "movieImage": "img/the-grand-budapest-hotel-poster.jpg",
            "moviePreview": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "movieRatingCount": "349 ratings",
            "movieRatingLevel": "Bad",
            "movieRatingScore": "3",
            "movieRunTime": 9900,
            "movieStarring": Array [
              "Jude Law",
              "Willem Dafoe",
            ],
            "movieVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            "title": "Dardjeeling Limited",
          },
        ]
      }
      tabs={
        Array [
          Object {
            "id": "TAB1",
            "isActive": true,
            "name": "Tab1",
          },
          Object {
            "id": "TAB2",
            "isActive": false,
            "name": "Tab2",
          },
          Object {
            "id": "TAB3",
            "isActive": false,
            "name": "Tab3",
          },
        ]
      }
    />
  </BrowserRouter>
</Context.Provider>
`;
